### ISSUE TITLE
Add syntax for declaring external record or scalar

### ISSUE THREAD

**Issue by cmyr** [googlefonts/fontations#91]
With the removal of `BigEndian`, we no longer have a mechanism whereby the user can communicate in a codegen file whether an unknown type is a scalar or a record. (It is currently possible to use unknown types that are tables, since tables are always referenced via an offset.)

This is a *fairly* uncommon case, but it does exist, and should be supported.

The information we will need to be able to express for these types is the following:
- The name of the type in the read crate
- The name of the type in the write crate (these will generally be the same?)
- possibly metadata/attributes that explain how to read the type? ~For instance if we're declaring a custom record, it is probably because it is non-trivial, which probably means it needs external arguments in order to be parsed. For instance `ValueRecord` requires a `ValueFormat` argument. ~ (Actually this is unnecessary? you'll need to have this metadata at the place where the type is used, but not at the declaration site. It's possible that other attributes may be required, though?)

What should syntax like this look like?

I think the best approach is to match as closely as possible the existing syntax. I think we can do this by introducing a new keyword, `extern` (which is already a rust keyword, although it is no longer used very often).

So basically I think that `ValueRecord` should be declared as,

```rust
extern record ValueRecord;
```

And a custom scalar (let's imagine `ValueFormat` required a custom implementation) would look like,

```rust
extern scalar ValueFormat;
```

And if required, we could add attributes to this.

At parse time, I think we would probably want to parse this into a new category of top-level item.

**Comment by rsheeter** [comment]
Once available, update parsing.rs to remove `is_wellknown_struct`; it isa workaround for inability to say ValueRecord is is extern
